pkg/webui/console/store/index.js

```diff
@@ -12,20 +12,88 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { mapValues } from 'lodash'
+import * as Sentry from '@sentry/browser'
 import { createStore, applyMiddleware, compose } from 'redux'
 import { createLogicMiddleware } from 'redux-logic'
 import { routerMiddleware } from 'connected-react-router'
+import createSentryMiddleware from 'redux-sentry-middleware'
 
 import dev from '@ttn-lw/lib/dev'
+import env from '@ttn-lw/lib/env'
 
 import createRootReducer from './reducers'
 import requestPromiseMiddleware from './middleware/request-promise-middleware'
 import logics from './middleware/logics'
 
+if (env.sentryDsn)
+  Sentry.init({
+    dsn: env.sentryDsn,
+    release: process.env.VERSION,
+    normalizeDepth: 10,
+  })
+
 const composeEnhancers = (dev && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose
 
 export default function(history) {
   const middleware = applyMiddleware(
+    createSentryMiddleware(Sentry, {
+      actionTransformer: action => {
+        switch (action.type) {
+          case 'GET_API_KEY_LIST_SUCCESS':
+          case 'GET_API_KEY_SUCCESS':
+            return {
+              ...action,
+              payload: {
+                ...action.payload,
+                entities: undefined,
+              },
+            }
+          default:
+            return action
+        }
+      },
+      stateTransformer: state => {
+        return {
+          ...state,
+          apiKeys: undefined,
+          collaborators: {
+            entities: {
+              ...state.collaborators.entities,
+              admin: {
+                ...state.collaborators.entities.admin,
+                ids: undefined,
+              },
+            },
+          },
+          devices: {
+            ...state.devices,
+            entities: mapValues(state.devices.entities, value => {
+              if (Boolean(value.ids)) {
+                return { ...value, ids: undefined }
+              }
+            }),
+          },
+          gateways: {
+            ...state.gateways,
+            entities: mapValues(state.gateways.entities, value => {
+              if (Boolean(value.ids)) {
+                return { ...value, ids: undefined }
+              }
+            }),
+          },
+          pagination: undefined,
+          user: {
+            ...state.user,
+            user: {
+              ...state.user.user,
+              ids: undefined,
```

Comment:
Sentry offers a way to remove the user id while retaining a correlatable ID, see: https://docs.sentry.io/data-management/sensitive-data/#examples

Reply:
in relation to this, `user_id` does not seem to be Personally Identifiable Information, so I don't think we should scrub this.

Reply:
Not per se but it can be,
1. if e.g. a user chooses to use his/her full name as ID
2. once it is correlated to our user account table